knex/index.d.ts

```diff
@@ -251,10 +259,12 @@ declare namespace Knex {
         (columnName: string, direction?: string): QueryBuilder;
     }
 
+    type UnionFunction = (this: QueryBuilder) => void;
+
     interface Union {
-        (callback: Function, wrap?: boolean): QueryBuilder;
-        (callbacks: Function[], wrap?: boolean): QueryBuilder;
-        (...callbacks: Function[]): QueryBuilder;
+        (callback: UnionFunction | UnionFunction[], wrap?: boolean): QueryBuilder;
+        (...callbacks: UnionFunction[]): QueryBuilder;
+        // no way to specifiy this siganiture in the current typescript version
```

Comment:
the comment is a note above:

// (...callbacks: Function[], wrap?: boolean): QueryInterface;

Which allows the specification of an arbitrary number of callbacks followed by an optional parameter to wrap them. Both spread operator arguments and optional arguments must be specified last, which as far as I know is not possible with typescript.

Docs for this function http://knexjs.org/#Builder-union